Add machine-local configuration option

cg505 opened the pull request:
Allows the user to add a machine-local config at `$XDG_CONFIG_HOME/dot/init.el`. If `$XDG_CONFIG_HOME` is not defined, it defaults to `~/.config/`.

*Closes #30*.

cg505 committed:
adds ability to machine-local config at ~/.config/dot/emacs/init.el

rye commented on .emacs.d/lisp/user/machine-local.el:
```diff
@@ -0,0 +1,5 @@
+(let ((dot-config (concat (file-name-directory (or (getenv "XDG_CONFIG_HOME") "~/.config/")) "dot/"))
```
Perhaps put this `let` in a `defun` or something like that?

rye replied:
Just so that we are able to perhaps extend this functionality in the future, that is.

rye reviewed (commented):
LGTM, and is pretty light-weight, but maybe make this a defun?

cg505 commented:
ok. can't get to it rn, so if you want to do it, go ahead

rye committed:
Extract chunks of behavior out into functions

These are somewhat simpler.  I'm not sure what the effect on
performance will be, but I'm sure this will work pretty decently well.

rye committed:
Refactor defuns to fix some problems

rye reviewed (approved):
This now looks good to me.  We might consider using `file-name-directory` a bit more semantically, but for now this looks great.
